terminal/TerminalClient.cpp

```diff
@@ -130,7 +137,139 @@ void handleWindowChanged(winsize* win) {
   }
 }
 
-int main(int argc, char** argv) {
+void initSetupSSH() {
+  string CLIENT_TERM(getenv("TERM"));
+  if (FLAGS_passkey.empty()) {
+    FILE *passkey_p = popen(
+        "env LC_ALL=C tr -dc \"a-zA-Z0-9\" < /dev/urandom | head -c 32", "r");
+    if (!passkey_p) {
+      cout << "cannot generate passkey!" << endl;
+      exit(1);
+    }
+    char passkey_buffer[1024];
+    fgets(passkey_buffer, sizeof(passkey_buffer), passkey_p);
+    pclose(passkey_p);
+    FLAGS_passkey = string(passkey_buffer);
+  }
+  if (FLAGS_id.empty()) {
+    FILE *id_p = popen(
+        "env LC_ALL=C tr -dc \"a-zA-Z0-9\" < /dev/urandom | head -c 16", "r");
+    if (!id_p) {
+      cout << "cannot generate id!" << endl;
+      exit(1);
+    }
+    char id_buffer[1024];
+    fgets(id_buffer, sizeof(id_buffer), id_p);
+    pclose(id_p);
+    FLAGS_id = string(id_buffer);
+  }
+
+
+  string cmdoptions = "--idpasskey=" + FLAGS_id + "/" + FLAGS_passkey;
+  string SSH_SCRIPT_PREFIX{
+      "SERVER_TMP_DIR=${TMPDIR:-${TMP:-${TEMP:-/tmp}}};"
+      "TMPFILE=$(mktemp $SERVER_TMP_DIR/et-server.XXXXXXXXXXXX);"
+      "PASSKEY="+FLAGS_passkey+";"
+      "ID="+FLAGS_id+";"
+      "export TERM="+CLIENT_TERM+";"
+      "/usr/bin/etserver "};
```

Comment:
we can't assume that people have etserver installed to /usr/bin.  Does putting etserer by itself fail when connecting to os/x?  If so, we need to run the ssh commands in interactive mode so the bashrc/zshrc files are executed.